### PR TITLE
tidy up smap converter

### PR DESCRIPTION
**Pull request by BenjaminRuston**
## Description

This PR removes files and renames testing files.  See `Files changed` for all details to summarize:

- [x] remove `src/land/smap9km_ssm2ioda.py` 
- [x] update CMakeLists
- [x] move `smap_ssm.nc` to `smap_nrt_ssm.nc`
- [x] move `smap9km_ssm.nc` to `smap_ssm.nc`

## Issue(s) addressed

Resolves #1514 

## Impact

This is a lateral move, and should cause no impact.

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have run the unit tests before creating the PR

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -2244,17 +2244,17 @@ ecbuild_add_test( TARGET  test_${PROJECT_NAME}_ghcn_snod
...
+ecbuild_add_test( TARGET  test_${PROJECT_NAME}_smap_nrt_ssm
                     TYPE    SCRIPT
                     ENVIRONMENT "PYTHONPATH=${IODACONV_PYTHONPATH}"
                     COMMAND bash
                     ARGS    ${CMAKE_BINARY_DIR}/bin/iodaconv_comp.sh
                             netcdf
                             "${Python3_EXECUTABLE} ${CMAKE_BINARY_DIR}/bin/smap_ssm2ioda.py
                             -i testinput/SMAP_L2_SM_P_NRT_24342_A_20190822T224858_N16023_002.h5
-                            -o testrun/smap_ssm.nc
+                            -o testrun/smap_nrt_ssm.nc
```
Comment (PatNichols) [inline test/CMakeLists.txt:2255]: @BenjaminRuston This file was not found in the test. Perhaps remove the test until we can get data?

Comment (BenjaminRuston) [reply test/CMakeLists.txt:2255]: the `testrun` is the ctest output... here the issue was with the `testoutput` and needed a change in the CMakeLists.txt

**Comment by BenjaminRuston** [comment]
@ClaraDraper-NOAA and @YoulongXia-NOAA this is `ready to go` it is a complete lateral move removing the deprecated file and renaming the test files in the ctests

**Review by ClaraDraper-NOAA — APPROVED** [review]
Looks good to me. Thanks @BenjaminRuston

**Review by YoulongXia-NOAA — APPROVED** [review]
Look good. I approve this PR.

**Pull request merged**